### PR TITLE
Bug 1268994 - Import en search engines from Android and remove en override

### PR DESCRIPTION
**Pull request by thebnich**
Since it seems like we've been falling behind recently with Android's en-US plugins, this seems like a good time to finally support en-US engine imports so we no longer have to maintain our own clones. All of the iOS-specific parameters we had are now included as overlays -- but please double-check to make sure I didn't miss anything!

### CODE REVIEW

File: Client/Assets/Search/SearchPlugins/en/yahoo.xml [hunk 1]
```diff
@@ -5,18 +5,25 @@
...
 <Url type="application/x-suggestions+json" method="GET" template="https://search.yahoo.com/sugg/ff">
-    <Param name="output" value="fxjson"/>
-    <Param name="appid" value="ffm"/>
-    <Param name="command" value="{searchTerms}"/>
-    <Param name="nresults" value="8"/>
+  <Param name="output" value="fxjson"/>
+  <Param name="appid" value="ffm"/>
+  <Param name="command" value="{searchTerms}"/>
+  <Param name="nresults" value="4"/>
```
Comment (flodolo) [inline Client/Assets/Search/SearchPlugins/en/yahoo.xml]: Is this change wanted or needs an overlay?

File: Client/Assets/Search/scrape_plugins.py [hunk 2]
```diff
@@ -14,7 +14,18 @@
 import subprocess
 import urllib
 
-header = """\
+# Paths for en-US plugins included in the core Android repo.
+EN_PLUGINS_DIR_URL = "https://hg.mozilla.org/releases/mozilla-aurora/file/tip/mobile/locales/en-US/searchplugins"
```
Comment (flodolo) [inline Client/Assets/Search/scrape_plugins.py]: tip is generally safe on mozilla-aurora, but for consistency you should use 'default' (don't try that on beta though, tip and default are two different beasts)

File: Client/Assets/Search/SearchPlugins/en/yahoo.xml (hunk 1 above)
Comment (thebnich) [inline Client/Assets/Search/SearchPlugins/en/yahoo.xml]: I think this change is fine -- it corresponds to the number of results shown in suggestions. We limit that number to 4 anyway in the UI, so this should have no effect. Also worth noting that Android used to have `nresults` set to 8 when we originally pulled these in, so this was not an iOS-specific change.

File: Client/Assets/Search/scrape_plugins.py (hunk 2 above)
Comment (thebnich) [inline Client/Assets/Search/scrape_plugins.py]: Thanks -- I always forget to change this.

**Pull request merged**